packages/compass-preferences-model/src/utils.ts

```diff
@@ -3,14 +3,12 @@ import type { ParsedGlobalPreferencesResult } from '.';
 import { setupPreferences } from './setup-preferences';
 import { UserStorage } from './storage';
 import { getStoragePaths } from '@mongodb-js/compass-utils';
-const { basepath } = getStoragePaths() || {};
-
-const userStorage = new UserStorage(basepath);
```

Comment:
This is the main reason that triggered this PR. Here, I instantiated `userStorage` ahead of time and the `basepath` at this point is `Electron` (default path). Once the appName is setup, this code is never run again and the old one persists and leads to incorrect path. 

Further in this PR, i changed this in all the places so we don't have `basepath` at the top level.

@mcasimir @kmruiz

Reply:
Something to think about for the refactor .. is there a way to get `getStoragePaths()` to always return the right result independently to where or when is called? It still feels like we could get a regression for this issue .. unless there is an easy way to test it 🤷‍♂️ ? .. in which case, that would be also a good idea.

Reply:
I think the idea initially was that this all happens in `main/application.ts`, where `this.setupUserDirectory()` is the first thing that happens, but now main/index.ts handles a ton of other stuff so the pattern broke and we are running into things like that. I don't think we can completely remove the need to have a place where setup happens and order is important, but we definitely should think about how to enforce that this setup happens in one place so that we don't run into issues like that

Reply:
Having a test would be so awesome. 

>  but we definitely should think about how to enforce that this setup happens in one place so that we don't run into issues like that.

I don't think so .. at least here I see a cohesion issue. If we would have a common place where all the fs / user data code is encapsulated that would never happen, just as an example:

```
class CommonClassToAccessUserData {
   private _setupUserDirectory(): void {
     if(this.setupDone) return;
     this.setupDone = true;
    if (process.env.NODE_ENV === 'development') {
      const appName = app.getName();
      // When NODE_ENV is dev, we are probably running the app unpackaged
      // directly with Electron binary which causes user dirs to be just
      // `Electron` instead of app name that we want here
      app.setPath('userData', path.join(app.getPath('appData'), appName));

      // @ts-expect-error this seems to work but not exposed as public path and
      // so is not available in d.ts files. As this is a dev-only path and
      // seemingly nothing is using this path anyway, we probably can ignore an
      // error here
      app.setPath('userCache', path.join(app.getPath('cache'), appName));
    } 
     
    async getStoragePathAndCreateParent(relativeFilePath) {
        this._setupUserDirectory();
        
        absoluteFilePath = path.join(userDir, relativeFilePath)
        mkdir(parentDir(absoluteFilePath), {recursive})
        
        return absoluteFilePath;
    }
    
    writeToFile(relativeFilePath, content) {
        fs.writeFile(await getStoragePathAndEnsureParent(), '...');
    }
}
```

Reply:
That said .. i don't know if we set the `appName` dynamically too, having a clear "please initialise everything here and in order" is also something that makes sense.

Reply:
Oh, totally, I think both things would be very good improvements to the codebase 🙂